### ISSUE TITLE
Potential bug in _fuse_leaves_by_immediate_parent()

### ISSUE THREAD

**Issue by nivlekp** [Abjad/abjad#1222]
As I am debugging the Nauert Quantizer, I came across this potential bug in the `_fuse_leaves_by_immediate_parent()` in the Abjad core.

If I am not mistaken, the following code should fuse the first five notes/leaves into `ef'4~ ef'16`

```
import abjad

voice = abjad.Voice(r"ef'16 ~ ef'16 ~ ef'16 ~ ef'16 ~ ef'16 r16 r16 r16 r4 r4")
logical_tie = voice[0]._get_logical_tie()
logical_tie._fuse_leaves_by_immediate_parent()
```

but instead, the following exception is raised:
`Exception: tie note or chord (not {leaf!r}).`

I am just wondering if this is a bug? If so, I might go ahead and try to fix it.

**Comment by trevorbaca** [comment]
Hi @nivlekp yes that is definitely a bug.

Also, I've reorganized a bunch of stuff in Abjad 3.2, most of which is checked into the mainline now. (A few more changes are sitting in my dev branch). The equivalent 3.2-style code to trigger the bug you've found looks like this:
```
voice = abjad.Voice(r"ef'16 ~ ef'16 ~ ef'16 ~ ef'16 ~ ef'16 r16 r16 r16 r4 r4")
leaves = abjad._iterate._get_logical_tie_leaves(voice[0])
selection = abjad.select(leaves)
leaves = abjad.mutate._fuse_leaves_by_immediate_parent(selection)
```

Those fuse bugs are nasty. If you think you can fix it, go ahead. I'll try to port into 3.2 if you're successful.

**Comment by nivlekp** [comment]
Hi @trevorbaca, thank you for your reply.
Sorry I did not notice before that Abjad 3.2 has some major reorganisations.
I have already made an attempt fixing that in the pull request #1223 
I have also added this specific case in the test.
Please let me know if I need to make further changes.

**Comment by trevorbaca** [comment]
Hi @nivlekp, this is great! Thank you so much. (Also for the interval calculation fix!) An article deadline is keeping me busy until the end of the week, so it will take me a few days to get around to pulling and confirming everything. But everything looks great, and I'll confirm in a few days.

**Comment by nivlekp** [comment]
Hi @trevorbaca, thank you for your reply.
It was no problem at all! I have to fix this bug anyway for my own purpose -- Nauert Quantizer would not work properly if the leaves are not fusing properly.

**Comment by trevorbaca** [comment]
Hi @nivlekp, I'm looking at this again now, and I can confirm that it's still a bug. The bug seems quite specific, and is occurs if two conditions are met. First, the leaves-to-be-fused sum to a nonassignable duration (like 5/4). Second, the leaves-to-be-fused are FOLLOWED immediately by a leaf that can not be tied (ie, a rest or skip).

So it works to fuse these five notes:
```
voice = abjad.Voice(r"c'4 ~ c'4 ~ c'4 ~ c'4 ~ c'4")                                       
selection = voice[:]                                                                      
leaves = abjad.mutate._fuse_leaves_by_immediate_parent(selection)                         
abjad.f(voice)                                                                            ~                                                                                         
\new Voice
{
    c'1
    ~
    c'4
}
```

And it even works to fuse these five notes together with the following rest:
```
voice = abjad.Voice(r"c'4 ~ c'4 ~ c'4 ~ c'4 ~ c'4 r4")                                    
selection = voice[:]                                                                      
leaves = abjad.mutate._fuse_leaves_by_immediate_parent(selection)                         
abjad.f(voice)                                                                            ~                                                                                         
\new Voice
{
    c'1.
}
```

But what breaks is attempting to fuse just these fives notes WHEN they are followed by the rest (ie, without including the rest in the fuse operation); note the change in the `voice[:5]` selection:
```
voice = abjad.Voice(r"c'4 ~ c'4 ~ c'4 ~ c'4 ~ c'4 r4")                                    
selection = voice[:5]                                                                     
leaves = abjad.mutate._fuse_leaves_by_immediate_parent(selection)                         
abjad.f(voice)                                                                            ~                                                                                         
Traceback (most recent call last):
  File "/Users/trevorbaca/.abjad/tmp.py", line 11, in <module>
    leaves = abjad.mutate._fuse_leaves_by_immediate_parent(selection)
  File "/Users/trevorbaca/abjad/abjad/mutate.py", line 64, in _fuse_leaves_by_immediate_parent
    fused = _fuse(part)
  File "/Users/trevorbaca/abjad/abjad/mutate.py", line 33, in _fuse
    return _fuse_leaves(SELECTION)
  File "/Users/trevorbaca/abjad/abjad/mutate.py", line 53, in _fuse_leaves
    result = _set_leaf_duration(leaves[0], total_preprolated)
  File "/Users/trevorbaca/abjad/abjad/mutate.py", line 141, in _set_leaf_duration
    tie(logical_tie_leaves)
  File "/Users/trevorbaca/abjad/abjad/spanners.py", line 2242, in tie
    raise Exception(rf"tie note or chord (not {leaf!r}).")
Exception: tie note or chord (not Rest('r4')).
```

I'm working through your proposed fix over in #1223 (thanks, btw!) but initially the fix seems to break five or six other pytests.

I'll keep working tomorrow and tell you what I come up with.

**Comment by trevorbaca** [comment]
The error is in `_set_leaf_duration()`. More tomorrow.
